### ISSUE TITLE
2D convolution on CIFAR10 example shows poor results

### ISSUE THREAD

**Issue by scottyler89** [tflearn/tflearn#272]
I posted this in stackoverflow, but it may be more appropriate here.
https://stackoverflow.com/questions/38885226/2d-convolution-in-tflearn-cnn-trashes-learning-for-mnist-and-cifar-10-benchmarks

Essentially I just want to make sure there isn't a bug in the 2D convolution or maxpooling functions. Following the tutorial script here:
https://github.com/tflearn/tflearn/blob/master/examples/images/convnet_cifar10.py

I get completely random classifications (10% validation accuracy), all the way through 50 epochs. Commenting out the 2d convolution and max pooling steps though, it goes up to about 50% validation accuracy. To me this sounds intuitively wrong (as if the convolution function is somehow accidentally shuffling the input data), but perhaps this example is simply a poor classifier.  Is this reproducible for others, or is there something wrong with my tflearn or tensorflow builds?

**Comment by aymericdamien** [comment]
Could you try to remove ImageAugmentation? It might come from there. For some reason it is not working well for some people.

**Comment by scottyler89** [comment]
Same results unfortunately; 10% validation accuracy in every epoch. If I were to guess, I'd think there's a silent bug that is messing with the data somehow. I'm really not familiar enough with it to know though.

**Comment by scottyler89** [comment]
I also tried getting rid of the image processing steps, but with the same results.

**Comment by scottyler89** [comment]
Is there a control I can run with a known theoretical output to see if my build can recapitulate the results from someone else's build? Or maybe just see the output of the 2d deconvolution or maxpooling steps directly? Thanks!

**Comment by aymericdamien** [comment]
Are you trying on GPU (old one)? I had issue with GTX770 before, but it was running well on other GPUs and CPU.

**Comment by scottyler89** [comment]
Oddly enough, that's the reverse of my situation. I just upgraded from GTX750 for a GTX1060. If I recall, I think the same script gave good results using the 750, but now isn't working on the 1060. It was a few weeks back though, and I didn't save any of the results from 750 training, so I'm not totally sure.

**Comment by aymericdamien** [comment]
Are you using latest driver, CUDA8 and latest CUDNN, and latest tensorflow? It may have an issue with those.
Could you also try on CPU?

**Comment by scottyler89** [comment]
I'm using the latest driver, but CUDA 7.5 and CUDNN v4 just because that's what it said on tensorflow's install site. If it seems to be working well with the latest versions, it might be worth trying the update. For running it on the CPU, is there a quick way to change that setting? I looked around briefly on the web, but just found the tflearn.config.init_graph function, but it still seemed to be running on the GPU when I allocated it some CPU in the num_cores, or if I changed the soft_placement arg.

**Comment by aymericdamien** [comment]
Yes, I think there is an issue with CUDA/CUDNN, if you can, better to update them, they will also give you performance boost.
to run on CPU, simply disable GPU devices using:
```
export CUDA_VISIBLE_DEVICES=''
python .....
```

**Comment by scottyler89** [comment]
Yes - indeed. I think you're right. Turned off the GPU, and saw the loss start to drop right away. Must be something in CUDA or CUDNN. Thanks so much for your help! I really appreciate it =)

**Comment by aymericdamien** [comment]
Cool, glad to hear that solved your issue
